kb/development/mbedtls-coding-standards.md

```diff
@@ -116,11 +116,17 @@ All public names (functions, variables, types, `enum` constants, macros) must st
     mbedtls_aes_setkey_decrypt()
 ```
 
+Note that all externally linked functions must have a name starting with `mbedtls_` to avoid link-time conflicts, even if they are not declared in a public header. This also applies to global variables (which should be used very sparingly).
+
 Exception: code implementing the PSA crypto API uses the `PSA_` and `psa_` prefixes. This includes official APIs as well as draft APIs that are on the PSA standards track. API extensions which are meant to remain specific to Mbed TLS, and internal functions, should use the `MBEDTLS_/mbedtls_` prefix, however there are many existing cases of using the `PSA_/psa_` prefix.
 
+Exception: macros may have lowercase names if they behave exactly like normal variables and functions. In particular, function-like macros may have lowercase names only if they expand to an expression that evaluates each argument exactly once.
```

Comment:
Remark: I understand that this reflects the reality at this point and therefore this paragraph should be here, so I don't suggest any changes. I just wanted to remark that I am not very happy about this exception and in the future I think we should tighten it (only for macros that evaluate to a function call) or eventually (after we reworked the platform abstraction layer) even remove this exception.

Reply:
I would suggest we write what we want to do, i.e. allow lower-case only for macros that evaluate to a function call (e.g. `#define mbedtls_calloc calloc`). There are always exceptions to the coding standards, and so writing something that reflects exactly what we have will means even having to capture inconsistencies. Writing what we would prefer to see seems better, IMO.

Reply:
I've updated the paragraph to say that while it isn't forbidden, it's unwanted.